src/avt/Database/Ghost/avtUnstructuredDomainBoundaries.C

```diff
@@ -218,22 +447,26 @@ avtUnstructuredDomainBoundaries::SetGivenCellsAndPoints(int fromDom, int toDom,
 
     if (index == -1)
     {
-        giveIndex.push_back(pair<int, int>(fromDom, toDom));
+        index = static_cast<int>(giveIndex.size());
+        giveIndex.emplace_back(fromDom, toDom);
         givenCells.push_back(cells);
-        givenPoints.resize(givenPoints.size() + 1);
-
-        sharedPointsMap.resize(sharedPointsMap.size() + 1);
+        // Use emplace_back to add a default-initialized std::vector<int> 
+        // to the vector, avoiding creating a temporary object.
+        givenPoints.emplace_back();
+        sharedPointsMap.emplace_back();
```

Comment:
@markcmiller86 here is a case where it is best to mix usage of `push_back` and `emplace_back`. The general rule is as follows:
- _Use emplace_back when you want to construct an object directly in the container._
- _Use push_back when you already have a fully constructed object that you want to add to the container._

In this case, `cells` is already constructed, so we must use `push_back`. But we can construct the other objects in place, so we use `emplace_back`.